fml/concurrent_message_loop.cc

```diff
@@ -159,15 +148,12 @@ void ConcurrentTaskRunner::PostTask(const fml::closure& task) {
     return;
   }
 
-  if (auto loop = weak_loop_.lock()) {
-    loop->PostTask(task);
-    return;
+  {
+    std::scoped_lock lock(weak_loop_mutex_);
```

Comment:
This adds a small amount of overhead compared to the shared_ptr, but we could share the `tasks_mutex_` if we wanted to add extra complexity.